pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -102,25 +114,40 @@ func (c *StaticResourceController) WithIgnoreNotFoundOnCreate() *StaticResourceC
 	return c
 }
 
+// WithConditionalResource makes the controller react to changes in dependent object
+// If the deleteConditionFunc returns true, delete the resource, if the conditionFunc returns true create the resource.
+// Deletion trumps creation. By default, deletionConditionFunc is !conditionFunc.
+// TODO: This can be made multi-resource aware as well.
+func (c *StaticResourceController) WithConditionalResource(createConditionFunc resourceapply.ConditionalFunction,
+	file string, deleteConditionFunc resourceapply.ConditionalFunction) *StaticResourceController {
+	rcm := resourceapply.ResourceConditionalMap{
```

Comment:
```go
conditionalManifests := (read them)
conditionalApplyFn := func() (directResourceResults, error){
  if deletionCondtionFn(){
    return resourceapply.DeleteAll(ctx, c.clients, syncContext.Recorder(), conditionalManifests), nil
  }
  if createConditionFn(){
    return resourceapply.ApplyDirectly(ctx, c.clients, syncContext.Recorder(), conditionalManifests), nil
  }
  return nil, nil
}
c.ConditionalApplyFns = append(conditionalApplyFn)

```

Then you add these below the existing ApplyDirectly in Sync.

Reply:
@deads2k - I slightly tweaked your suggestion and applied the changes. PTAL when you find time.